<h1>UK ACCIDENT REPORT DATA ANALYSIS</h1>

<h3>ANALYST: DENNIS KARL S. DOMANAIS</h3>

<h1>IMPORTING LIBRARIES NEEDED</h1>

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import f_oneway

<h1>IMPORTING DATASETS</h1>

In [2]:
accident = pd.read_csv("accident_data.csv")

In [3]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [4]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [5]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [6]:
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna('unknown road condition')
accident['Road_Type'] = accident['Road_Type'].fillna('unaccounted')
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna('unaccounted')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

<h1>CONVERT COLUMNS TO ITS APPROPRIATE DATA TYPE</h1>

In [7]:
accident['Index'] = accident['Index'].astype('category')
accident['Accident_Severity'] = accident['Accident_Severity'].astype('category')
accident['Light_Conditions'] = accident['Light_Conditions'].astype('category') 
accident['District Area'] = accident['District Area'].astype('category')
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].astype('category')
accident['Road_Type'] = accident['Road_Type'].astype('category')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].astype('category')
accident['Weather_Conditions'] = accident['Weather_Conditions'].astype('category')
accident['Vehicle_Type'] = accident['Vehicle_Type'].astype('category')

In [8]:
accident.dtypes

Index                      category
Accident_Severity          category
Accident Date                object
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

In [9]:
accident['Accident Date'] = accident['Accident Date'].astype('str')
accident['Accident Date'] = accident['Accident Date'].str.strip()
accident['Accident Date'] = accident['Accident Date'].str.replace('/','-')
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

In [10]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'],dayfirst =True, errors = 'coerce')
accident.dtypes

Index                            category
Accident_Severity                category
Accident Date              datetime64[ns]
Latitude                          float64
Light_Conditions                 category
District Area                    category
Longitude                         float64
Number_of_Casualties                int64
Number_of_Vehicles                  int64
Road_Surface_Conditions          category
Road_Type                        category
Urban_or_Rural_Area              category
Weather_Conditions               category
Vehicle_Type                     category
dtype: object

<h1>DETERMINING NULL VALUES AND FILLNG THEM</h1>

<h4>DETERMING NULLS</h4>

In [11]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

In [12]:
accident['Year']=accident['Accident Date'].dt.year
accident['Month']=accident['Accident Date'].dt.month
accident['Day']=accident['Accident Date'].dt.day
accident['DayofWeek']=accident['Accident Date'].dt.dayofweek
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious    2019-08-26  51.517715   
3       200701BS66128           Serious    2019-08-16  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    2022-02-18  57.374005   
660675  201091NM01881            Slight    2022-02-21  57.232273   
660676  201091NM01935            Slight    2022-02-23  57.585044   
660677  201091NM01964           Serious    2022-02-23  57.214898   
660678  201091NM02142           Serious    2022-02-28  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2              unaccounted               Urban            unaccounted   
3       Single carriageway               Urban     Fine no high winds   
4              unaccounted               Urban            unaccounted   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  Year  Month  Day  DayofWeek  
0                                        Car  2019      6    5          2  
1                                        Car  2019      7    2          1  
2                      Taxi/Private hire car  2019      8   26          0  
3       Bus or coach (17 or more pass seats)  2019      8   16          4  
4                              Other vehicle  2019      9    3          1  
...                                      ...   ...    ...  ...  

In [13]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
DayofWeek                  0
dtype: int64

<h1 style="background-color:#252525; height:500; font-size:100; color:orange; font-weight:bold; padding:10px; text-align:center;">INSIGHTS</h1>

<h1>INSIGHT #1</h1>

<h3>What vehicle type is most frequently involved in accidents?</h3>

<h3 style="background-color:yellow;">The most common or dominant vehicle type involved in accidents is Car, indicating a higher likelihood of road incidents compared to other vehicle types.</h3>
<hr>

In [14]:
vehicle_types = accident['Vehicle_Type'].value_counts().reset_index()
vehicle_types

Vehicle_Type   count
0                                     Car  497992
1     Van / Goods 3.5 tonnes mgw or under   34160
2    Bus or coach (17 or more pass seats)   25878
3                   Motorcycle over 500cc   25657
4           Goods 7.5 tonnes mgw and over   17307
5              Motorcycle 125cc and under   15269
6                   Taxi/Private hire car   13294
7   Motorcycle over 125cc and up to 500cc    7656
8               Motorcycle 50cc and under    7603
9         Goods over 3.5t. and under 7.5t    6096
10                          Other vehicle    5637
11       Minibus (8 - 16 passenger seats)    1976
12                   Agricultural vehicle    1947
13                            Pedal cycle     197
14           Data missing or out of range       6
15                           Ridden horse       4

<h1>INSIGHT #2</h1>

<h3>On which road type do most car accidents occur?</h3>

<h3 style="background-color:yellow;">Most car accidents occur on single carriageways, likely due to high traffic volume, lack of dividers, and frequent intersections.</h3>

In [15]:
car_accidents = accident[accident['Vehicle_Type'] == 'Car']
road_type_counts = car_accidents['Road_Type'].value_counts()
road_type_counts

Road_Type
Single carriageway    371028
Dual carriageway       74820
Roundabout             33187
One way street         10223
Slip road               5298
unaccounted             3436
Name: count, dtype: int64

<h1>INSIGHT #3</h1>

<h3 style="background-color:yellow;">Most car accidents occur on dry roads, suggesting that factors was the Driver, like overspeeding or distracted.</h3>


In [70]:
road_surface= car_accidents['Road_Surface_Conditions'].value_counts()
road_surface


Road_Surface_Conditions
Dry                       337311
Wet or damp               140764
Frost or ice               14108
Snow                        4483
Flood over 3cm. deep         777
unknown road condition       549
Name: count, dtype: int64

<h1>INSIGHT #4</h1>

<h3 style="background-color:yellow;">Most car accidents occur in fine weather with no high winds, suggesting that human error, such as distraction, speeding, or reckless driving, plays a bigger role in accidents than adverse weather conditions.</h3>

In [71]:
weather = car_accidents['Weather_Conditions'].value_counts()
weather


Weather_Conditions
Fine no high winds       392668
Raining no high winds     59940
Other                     12993
unaccounted               10656
Raining + high winds       7206
Fine + high winds          6463
Snowing no high winds      4748
Fog or mist                2641
Snowing + high winds        677
Name: count, dtype: int64

<h1>INSIGHT #5</h1>

<h3>What vehicle type has the highest number of Fatal, Serious, and Slight accidents?</h3>

<h3 style="background-color:yellow;">Cars have the highest number of Fatal, Serious, and Slight accidents, indicating that they are the most frequently involved vehicle type in road accidents.</h3>

In [72]:
fatal_accidents = accident[accident['Accident_Severity'] == 'Fatal']
fatal_vehicle = fatal_accidents['Vehicle_Type'].value_counts()
fatal_vehicle



Vehicle_Type
Car                                      6577
Van / Goods 3.5 tonnes mgw or under       467
Motorcycle over 500cc                     339
Bus or coach (17 or more pass seats)      325
Goods 7.5 tonnes mgw and over             216
Motorcycle 125cc and under                189
Taxi/Private hire car                     155
Motorcycle over 125cc and up to 500cc     105
Motorcycle 50cc and under                  95
Other vehicle                              70
Goods over 3.5t. and under 7.5t            67
Minibus (8 - 16 passenger seats)           29
Agricultural vehicle                       21
Pedal cycle                                 6
Data missing or out of range                0
Ridden horse                                0
Name: count, dtype: int64

In [73]:
fatal_accidents = accident[accident['Accident_Severity'] == 'Slight']
fatal_vehicle = fatal_accidents['Vehicle_Type'].value_counts()
fatal_vehicle


Vehicle_Type
Car                                      424954
Van / Goods 3.5 tonnes mgw or under       29139
Bus or coach (17 or more pass seats)      22180
Motorcycle over 500cc                     21861
Goods 7.5 tonnes mgw and over             14770
Motorcycle 125cc and under                13049
Taxi/Private hire car                     11368
Motorcycle over 125cc and up to 500cc      6537
Motorcycle 50cc and under                  6494
Goods over 3.5t. and under 7.5t            5172
Other vehicle                              4800
Minibus (8 - 16 passenger seats)           1671
Agricultural vehicle                       1644
Pedal cycle                                 152
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

In [74]:
fatal_accidents = accident[accident['Accident_Severity'] == 'Serious']
fatal_vehicle = fatal_accidents['Vehicle_Type'].value_counts()
fatal_vehicle


Vehicle_Type
Car                                      66461
Van / Goods 3.5 tonnes mgw or under       4554
Motorcycle over 500cc                     3457
Bus or coach (17 or more pass seats)      3373
Goods 7.5 tonnes mgw and over             2321
Motorcycle 125cc and under                2031
Taxi/Private hire car                     1771
Motorcycle 50cc and under                 1014
Motorcycle over 125cc and up to 500cc     1014
Goods over 3.5t. and under 7.5t            857
Other vehicle                              767
Agricultural vehicle                       282
Minibus (8 - 16 passenger seats)           276
Pedal cycle                                 39
Data missing or out of range                 0
Ridden horse                                 0
Name: count, dtype: int64

<h1>INSIGHT #6</h1>

<h3>What year got the highest record of Accidents</h3>

<h3 style="background-color:yellow;">2019 got the Higher number or count of Accidents</h3>

In [75]:
car_yr = car_accidents['Year'].value_counts()
car_yr


Year
2019    130304
2021    128026
2020    127894
2022    111768
Name: count, dtype: int64

<h1>INSIGHT #7</h1>

<h3>What Year got the highest record of Fatal accidents?</h3>

<h3 style="background-color:yellow;">In Year 2019, It has the highest record of Fatal accidents</h3>




In [76]:
fatal_yr = accident[accident['Accident_Severity'] == 'Fatal'].groupby('Year').size()
fatal_yr

Year
2019    2714
2020    2341
2021    2057
2022    1549
dtype: int64

<h1>INSIGHT #8</h1>

<h3>Which District Area has the highest number of recorded accidents?</h3>

<h3 style="background-color:yellow;">The results show that Birmingham has the highest number of recorded accidents.</h3>


In [24]:
accident['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<h1>INSIGHT #9</h1>

<h3>Which accident severity has the highest record?</h3>

<h3 style="background-color:yellow;">Among all recorded accidents, Slight accidentshave the highest occurrence. This indicates that while accidents are frequent, most of them result in minor injuries or damage rather than severe or fatal outcomes. </h3>

In [77]:
birmingham_sev= accident["Accident_Severity"].value_counts()
birmingham_sev


Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h1>INSIGHT #10</h1>

<h3>On what day got of week has the highest count od accidents</h3>

<h3 style="background-color:yellow;">Accidents occur more frequently on weekends, especially on Saturdays, which have the highest recorded number of accidents. This trend could be due to increased travel, social activities, and higher traffic volumes during the weekend, leading to a greater risk of collisions.</h3>

In [78]:
birm_dw = accident["DayofWeek"].value_counts()
week = {
    0: 'Monday',
    1: 'Tuesday',
    2: 'Wednesday',
    3: 'Thursday',
    4: 'Friday',
    5: 'Saturday',
    6: 'Sunday',
}

birm_dw.index = birm_dw.index.map(week)
birm_dw


DayofWeek
Saturday     107178
Wednesday     99558
Thursday      99511
Friday        97900
Tuesday       94550
Sunday        89302
Monday        72680
Name: count, dtype: int64

<h1>INSIGHT #11</h1>

<h3>How many vehicles are usually involved in accidents?</h3>

<h3 style="background-color:yellow;">Most accidents involved two vehicles.</h3>

In [79]:
vehi = accident["Number_of_Vehicles"].value_counts()
vehi


Number_of_Vehicles
2     391995
1     200787
3      52806
4      11300
5       2464
6        758
7        296
8        147
9         57
10        27
11        12
14         9
12         7
13         6
16         4
28         1
15         1
32         1
19         1
Name: count, dtype: int64

<h1>INSIGHT #12</h1>

<h3>Two vehicles are commonly involved in accidents. What is the percentage distribution of Fatal, Serious, and Slight accidents when two vehicles are involved?</h3>

<h3 style="background-color:yellow;">Out of 391,995 accidents involving two vehicles, most were Slight at 88.5%, followed by Serious at 10.6%, while Fatal accidents had the lowest percentage at 0.9%.</h3>


In [81]:
vehicle_accidents = accident[accident["Number_of_Vehicles"] == 2]
severity_counts = vehicle_accidents["Accident_Severity"].value_counts()

severity_counts


Accident_Severity
Slight     346950
Serious     41578
Fatal        3467
Name: count, dtype: int64

<h1>INSIGHT #13</h1>

<h3>How does accident severity vary between urban and rural areas?</h3>

<h3  style="background-color: yellow">Rural areas have a higher number of fatal accidents compared to urban areas, despite urban areas having more overall accidents. This suggests that higher speeds, fewer traffic controls, and emergency response times in rural areas may contribute to increased fatality rates.</h3>

In [83]:
ur_severity = accident.groupby('Urban_or_Rural_Area')['Accident_Severity'].value_counts().unstack()
ur_severity



C:\Users\DLL-AccountOfficer\AppData\Local\Temp\ipykernel_7524\4000230851.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  ur_severity = accident.groupby('Urban_or_Rural_Area')['Accident_Severity'].value_counts().unstack()


Accident_Severity    Fatal  Serious  Slight
Urban_or_Rural_Area                        
Rural                 5601    37312  196077
Unallocated              0        1      10
Urban                 3060    50904  367714

<h1>INSIGHT #14</h1>

<h3>How do road surface conditions impact accidents in urban and rural areas?</h3>

<h3  style="background-color: yellow">Most accidents occur on dry roads in both urban and rural areas, but wet or damp conditions significantly contribute to accidents as well. Rural areas see a higher number of accidents on frosty or icy roads compared to urban areas, likely due to less frequent road maintenance and salting. This highlights the need for caution in adverse weather conditions, especially in rural locations.</h3>

In [63]:
ruban_sur = accident.groupby('Urban_or_Rural_Area')['Road_Surface_Conditions'].value_counts().unstack()
ruban_sur

C:\Users\DLL-AccountOfficer\AppData\Local\Temp\ipykernel_7524\2753767658.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  ruban_sur = accident.groupby('Urban_or_Rural_Area')['Road_Surface_Conditions'].value_counts().unstack()


Road_Surface_Conditions     Dry  Flood over 3cm. deep  Frost or ice  Snow  \
Urban_or_Rural_Area                                                         
Rural                    144861                   785         10953  3102   
Unallocated                   5                     1             0     0   
Urban                    302955                   231          7564  2788   

Road_Surface_Conditions  Wet or damp  unknown road condition  
Urban_or_Rural_Area                                           
Rural                          79005                     284  
Unallocated                        5                       0  
Urban                         107698                     442

<h1>INSIGHT #15</h1>

<h3>How do light conditions impact accident severity?</h3>

<h3 style="background-color:yellow"> Most fatal accidents occur in daylight, but a significant number also happen in complete darkness without lighting. Serious and slight accidents are most frequent in well-lit conditions, possibly due to higher traffic volume during those times. This suggests that while lighting improves visibility, it does not eliminate accidents, and darkness without lighting increases the risk of severe crashes.</h3>

In [85]:
sev_light_casual = accident.groupby(['Accident_Severity', 'Light_Conditions'])['Number_of_Casualties'].sum().unstack()
sev_light_casual


C:\Users\DLL-AccountOfficer\AppData\Local\Temp\ipykernel_7524\3309597506.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  sev_light_casual = accident.groupby(['Accident_Severity', 'Light_Conditions'])['Number_of_Casualties'].sum().unstack()


Light_Conditions   Darkness - lighting unknown  Darkness - lights lit  \
Accident_Severity                                                       
Fatal                                      114                   3421   
Serious                                   1138                  27847   
Slight                                    7219                 145840   

Light_Conditions   Darkness - lights unlit  Darkness - no lighting  Daylight  
Accident_Severity                                                             
Fatal                                   76                    3362      9510  
Serious                                518                   12913     87023  
Slight                                2791                   41440    553356

<h1>INSIGHT #16</h1>

<h3> How do accident counts vary across different months and years?</h3>

<h3 style="background-color:yellow">November consistently had the highest number of accidents each year, with 2019 recording the highest count (16,559). Accident counts tend to peak in late autumn and summer months, possibly due to weather conditions, increased travel, or seasonal factors. Notably, there is a steady decline in overall accident counts from 2019 to 2022, which may be influenced by changes in road regulations, pandemic-related restrictions, or improved safety measures. </h3>

In [66]:
mont_yr = accident.groupby(['Year'])['Month'].value_counts().reset_index(name='Count')
mon = {
    1: 'January',
    2: 'February',
    3: 'March',
    4: 'April',
    5: 'May',
    6: 'June',
    7: 'July',
    8: 'August',
    9: 'September',
    10: 'October',
    11: 'November',
    12: 'Decemeber'
}
mont_yr['Month'] = mont_yr['Month'].map(mon)
mont_yr


Year      Month  Count
0   2019   November  16559
1   2019       July  15862
2   2019        May  15833
3   2019       June  15528
4   2019    October  15528
5   2019    January  15355
6   2019  September  15271
7   2019      March  15049
8   2019     August  15044
9   2019  Decemeber  14708
10  2019      April  14125
11  2019   February  13253
12  2020    October  15684
13  2020   November  14770
14  2020       July  14630
15  2020  September  14432
16  2020   February  14353
17  2020        May  14336
18  2020       June  14205
19  2020    January  14133
20  2020  Decemeber  13794
21  2020      March  13494
22  2020      April  13394
23  2020     August  13366
24  2021   November  15473
25  2021    October  14834
26  2021       July  14300
27  2021       June  13936
28  2021        May  13811
29  2021  September  13792
30  2021  Decemeber  13709
31  2021    January  13417
32  2021     August  13415
33  2021      March  13202
34  2021      April  12715
35  2021   February  10950
36  2022   November  13622
37  2022    October  13534
38  2022  September  12960
39  2022       June  12812
40  2022       July  12653
41  2022        May  12372
42  2022      March  12341
43  2022     August  12088
44  2022      April  11510
45  2022   February  10935
46  2022    January   9967
47  2022  Decemeber   9625

<h1>INSIGHT #17</h1>

<h3>How do different light conditions affect the number of accidents?</h3>

<h3 style="background-color:yellow">Most accidents happen in daylight (484,880), likely due to high traffic. Accidents in darkness with streetlights on (129,335) are also common. Fewer accidents occur in areas with no lighting (37,437) or unlit streetlights (2,543), showing that poor visibility increases risk. </h3>

In [41]:
accident['Light_Conditions'].value_counts()


Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<h1>INSIGHT #18</h1>

<h3>How does the number of accidents vary based on Light Conditions for each Month?</h3>

<h3 style="background-color: yellow">Accidents occur more frequently during daylight, but the number of accidents in darkness increases significantly in winter months (October to December). This suggests that shorter daylight hours and reduced visibility contribute to higher accident rates in darker conditions.</h3>

In [68]:
month_light = accident.groupby(['Month', 'Light_Conditions']).size().reset_index(name='Count')

mon = {
    1: 'January',
    2: 'February',
    3: 'March',
    4: 'April',
    5: 'May',
    6: 'June',
    7: 'July',
    8: 'August',
    9: 'September',
    10: 'October',
    11: 'November',
    12: 'Decemeber'
}


month_light['Month'] = month_light['Month'].map(mon)
month_light

C:\Users\DLL-AccountOfficer\AppData\Local\Temp\ipykernel_7524\3958258887.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  month_light = accident.groupby(['Month', 'Light_Conditions']).size().reset_index(name='Count')


Month             Light_Conditions  Count
0     January  Darkness - lighting unknown    882
1     January        Darkness - lights lit  17386
2     January      Darkness - lights unlit    314
3     January       Darkness - no lighting   5161
4     January                     Daylight  29129
5    February  Darkness - lighting unknown    672
6    February        Darkness - lights lit  13117
7    February      Darkness - lights unlit    229
8    February       Darkness - no lighting   4062
9    February                     Daylight  31411
10      March  Darkness - lighting unknown    469
11      March        Darkness - lights lit  10387
12      March      Darkness - lights unlit    202
13      March       Darkness - no lighting   2873
14      March                     Daylight  40155
15      April  Darkness - lighting unknown    321
16      April        Darkness - lights lit   6063
17      April      Darkness - lights unlit    151
18      April       Darkness - no lighting   1759
19      April                     Daylight  43450
20        May  Darkness - lighting unknown    287
21        May        Darkness - lights lit   5534
22        May      Darkness - lights unlit    150
23        May       Darkness - no lighting   1536
24        May                     Daylight  48845
25       June  Darkness - lighting unknown    283
26       June        Darkness - lights lit   4473
27       June      Darkness - lights unlit    133
28       June       Darkness - no lighting   1343
29       June                     Daylight  50249
30       July  Darkness - lighting unknown    290
31       July        Darkness - lights lit   4989
32       July      Darkness - lights unlit    142
33       July       Darkness - no lighting   1512
34       July                     Daylight  50512
35     August  Darkness - lighting unknown    331
36     August        Darkness - lights lit   6439
37     August      Darkness - lights unlit    151
38     August       Darkness - no lighting   2047
39     August                     Daylight  44945
40  September  Darkness - lighting unknown    394
41  September        Darkness - lights lit   8055
42  September      Darkness - lights unlit    170
43  September       Darkness - no lighting   2389
44  September                     Daylight  45447
45    October  Darkness - lighting unknown    655
46    October        Darkness - lights lit  13320
47    October      Darkness - lights unlit    250
48    October       Darkness - no lighting   3606
49    October                     Daylight  41749
50   November  Darkness - lighting unknown    972
51   November        Darkness - lights lit  20795
52   November      Darkness - lights unlit    349
53   November       Darkness - no lighting   5575
54   November                     Daylight  32733
55  Decemeber  Darkness - lighting unknown    928
56  Decemeber        Darkness - lights lit  18777
57  Decemeber      Darkness - lights unlit    302
58  Decemeber       Darkness - no lighting   5574
59  Decemeber                     Daylight  26255

<h1>INSIGHT #19</h1>

<h3>How do weather conditions affect accidents on different road types?</h3>

<h3 style="background-color: yellow">Most accidents occur on single carriageways, especially in fine weather, as they handle high traffic. Rain and fog increase accident risks due to slippery roads and reduced visibility. Snow-related accidents are fewer but still mainly happen on single carriageways.</h3>

In [69]:
weather_road = accident.groupby(['Weather_Conditions', 'Road_Type']).size().reset_index(name='Count')
weather_road


C:\Users\DLL-AccountOfficer\AppData\Local\Temp\ipykernel_7524\3495326850.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  weather_road = accident.groupby(['Weather_Conditions', 'Road_Type']).size().reset_index(name='Count')


Weather_Conditions           Road_Type   Count
0       Fine + high winds    Dual carriageway    1519
1       Fine + high winds      One way street     158
2       Fine + high winds          Roundabout     540
3       Fine + high winds  Single carriageway    6178
4       Fine + high winds           Slip road     109
5       Fine + high winds         unaccounted      50
6      Fine no high winds    Dual carriageway   76916
7      Fine no high winds      One way street   11057
8      Fine no high winds          Roundabout   34667
9      Fine no high winds  Single carriageway  389830
10     Fine no high winds           Slip road    5520
11     Fine no high winds         unaccounted    2895
12            Fog or mist    Dual carriageway     682
13            Fog or mist      One way street      29
14            Fog or mist          Roundabout     222
15            Fog or mist  Single carriageway    2541
16            Fog or mist           Slip road      33
17            Fog or mist         unaccounted      21
18                  Other    Dual carriageway    2264
19                  Other      One way street     309
20                  Other          Roundabout    1109
21                  Other  Single carriageway   13156
22                  Other           Slip road     180
23                  Other         unaccounted     132
24   Raining + high winds    Dual carriageway    2033
25   Raining + high winds      One way street     153
26   Raining + high winds          Roundabout     563
27   Raining + high winds  Single carriageway    6703
28   Raining + high winds           Slip road     107
29   Raining + high winds         unaccounted      56
30  Raining no high winds    Dual carriageway   13044
31  Raining no high winds      One way street    1429
32  Raining no high winds          Roundabout    5347
33  Raining no high winds  Single carriageway   58581
34  Raining no high winds           Slip road     865
35  Raining no high winds         unaccounted     430
36   Snowing + high winds    Dual carriageway     185
37   Snowing + high winds      One way street      14
38   Snowing + high winds          Roundabout      30
39   Snowing + high winds  Single carriageway     642
40   Snowing + high winds           Slip road      10
41   Snowing + high winds         unaccounted       4
42  Snowing no high winds    Dual carriageway    1020
43  Snowing no high winds      One way street      78
44  Snowing no high winds          Roundabout     290
45  Snowing no high winds  Single carriageway    4751
46  Snowing no high winds           Slip road      69
47  Snowing no high winds         unaccounted      30
48            unaccounted    Dual carriageway    1761
49            unaccounted      One way street     332
50            unaccounted          Roundabout    1224
51            unaccounted  Single carriageway    9761
52            unaccounted           Slip road     148
53            unaccounted         unaccounted     902

<h1>INSIGHT #20</h1>

<h3>How does road surface condition impact the severity of accidents?</h3>

<h3 style="background-color:yellow">Most accidents happen on dry roads, but wet or damp conditions significantly increase the severity of crashes. Ice, snow, and floods contribute to fewer accidents overall but have a higher risk of severe or fatal outcomes. Proper road maintenance and driver caution in poor weather conditions can help reduce accidents</h3>

In [49]:
road_severity = accident.groupby(['Road_Surface_Conditions', 'Accident_Severity']).size().reset_index(name='Count')
road_severity


C:\Users\DLL-AccountOfficer\AppData\Local\Temp\ipykernel_7524\3851822773.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  road_severity = accident.groupby(['Road_Surface_Conditions', 'Accident_Severity']).size().reset_index(name='Count')


Road_Surface_Conditions Accident_Severity   Count
0                      Dry             Fatal    5788
1                      Dry           Serious   61638
2                      Dry            Slight  380395
3     Flood over 3cm. deep             Fatal      23
4     Flood over 3cm. deep           Serious     152
5     Flood over 3cm. deep            Slight     842
6             Frost or ice             Fatal     193
7             Frost or ice           Serious    2007
8             Frost or ice            Slight   16317
9                     Snow             Fatal      35
10                    Snow           Serious     565
11                    Snow            Slight    5290
12             Wet or damp             Fatal    2620
13             Wet or damp           Serious   23785
14             Wet or damp            Slight  160303
15  unknown road condition             Fatal       2
16  unknown road condition           Serious      70
17  unknown road condition            Slight     654